server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java

```diff
@@ -358,6 +359,60 @@ public TabletServer(ServerConfigurationFactory confFactory, VolumeManager fs) th
     this.logSorter = new LogSorter(instance, fs, aconf);
     this.replWorker = new ReplicationWorker(this, fs);
     this.statsKeeper = new TabletStatsKeeper();
+    final int numBusyTabletsToLog = aconf.getCount(Property.TSERV_LOG_BUSY_TABLETS_COUNT);
+    final long logBusyTabletsDelay =
+        aconf.getTimeInMillis(Property.TSERV_LOG_BUSY_TABLETS_INTERVAL);
+
+    // This thread will calculate and log out the busiest tablets based on ingest count and
+    // query count every #{logBusiestTabletsDelay}
+    if (numBusyTabletsToLog > 0) {
+      SimpleTimer.getInstance(aconf).schedule(new Runnable() {
+        @Override
+        public void run() {
+          Comparator<Pair<String,Long>> busiestTabletComparator =
+              new Comparator<Pair<String,Long>>() {
+                @Override
+                public int compare(Pair<String,Long> first, Pair<String,Long> second) {
+                  return second.getSecond().compareTo(first.getSecond());
+                }
+              };
+          PriorityQueue<Pair<String,Long>> busiestTabletsByIngestCount =
+              new PriorityQueue<>(numBusyTabletsToLog, busiestTabletComparator);
+          PriorityQueue<Pair<String,Long>> busiestTabletsByQueryCount =
+              new PriorityQueue<>(numBusyTabletsToLog, busiestTabletComparator);
+          synchronized (onlineTablets) {
+            for (Tablet tablet : onlineTablets.values()) {
+              addToBusiestTablets(tablet.totalIngest(), busiestTabletsByIngestCount,
+                  numBusyTabletsToLog);
+              addToBusiestTablets(tablet.totalQueries(), busiestTabletsByQueryCount,
+                  numBusyTabletsToLog);
+            }
+            logBusyTablets(busiestTabletsByIngestCount, "QUERY", numBusyTabletsToLog);
+            logBusyTablets(busiestTabletsByQueryCount, "INGEST", numBusyTabletsToLog);
+          }
+        }
+
+        private void addToBusiestTablets(long count,
+            PriorityQueue<Pair<String,Long>> busiestTabletsQueue, int numBusiestTabletsToLog) {
+          if (busiestTabletsQueue.size() < numBusiestTabletsToLog
+              || busiestTabletsQueue.peek().getSecond() < count) {
```

Comment:
I think this priority queue has the minimum count at the top, it that correct?

When is something ever added to busiestTabletsQueue?

Reply:
doh!